sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConnection.cs

```diff
@@ -167,158 +160,184 @@ public EventHubClient(string connectionString) : this(connectionString, null, nu
         ///
-        /// <returns>The set of information for the Event Hub that this client is associated with.</returns>
+        public virtual void Close(CancellationToken cancellationToken = default) => CloseAsync(cancellationToken).GetAwaiter().GetResult();
+
+        /// <summary>
+        ///   Performs the task needed to clean up resources used by the <see cref="EventHubConnection" />,
+        ///   including ensuring that the connection itself has been closed.
+        /// </summary>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public virtual async ValueTask DisposeAsync() => await CloseAsync().ConfigureAwait(false);
+
+        /// <summary>
+        ///   Determines whether the specified <see cref="System.Object" /> is equal to this instance.
+        /// </summary>
+        ///
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        ///
+        /// <returns><c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => base.Equals(obj);
+
+        /// <summary>
+        ///   Returns a hash code for this instance.
+        /// </summary>
+        ///
+        /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        ///   Converts the instance to string representation.
+        /// </summary>
+        ///
+        /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
         ///
-        public virtual Task<EventHubProperties> GetPropertiesAsync(CancellationToken cancellationToken = default) => InnerClient.GetPropertiesAsync(cancellationToken);
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Retrieves information about the Event Hub that the connection is associated with, including
+        ///   the number of partitions present and their identifiers.
+        /// </summary>
+        ///
+        /// <param name="retryPolicy">The retry policy to use as the basis for retrieving the information.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information for the Event Hub that this connection is associated with.</returns>
+        ///
+        internal virtual Task<EventHubProperties> GetPropertiesAsync(EventHubRetryPolicy retryPolicy,
```

Comment:
They're accessed only because this is an interim step;  the producer and consumer haven't yet been refactored to not have partition affinity.   Once they are, these shouldn't be called on the connection by tests, rather they should be called on the producer or consumer.

As for the "how" - our internal methods are shared with tests.